src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -1501,6 +1501,10 @@ private static List<ProjectRestoreMetadataFrameworkInfo> ReadTargetFrameworks(Js
                             });
                         });
                     }
+                    else if(propertyName == "targetAlias")
+                    {
+                        frameworkGroup.TargetAlias = jsonReader.ReadNextTokenAsString();
```

Comment:
Does the restore code expect alias to always be set?

I think `TargetAlias` should be defaulted to `frameworkPropertyName`'s value up on line 1451, and overridden here if the json specified it.

Reply:
> Does the restore code expect alias to always be set?

No, only when a project is multi targeted

Reply:
well, only when a project is SDK based :D